http_request.c

```diff
@@ -0,0 +1,64 @@
+#include "http_request.h"
+#include <unistd.h>
+#include <stdio.h>
+
+int write_verb_path(const int fd, const char* verb, const char* path) {
+    return dprintf(fd, "%s %s %s\r\n", verb, path, HTTP_VERSION);
+}
+
+int write_host(const int fd, const char* host) {
+    return dprintf(fd, "HOST: %s\r\n", host);
+}
+
+int write_headers(const int fd, char* headers[], const int header_count) {
+    int i, err, sum = 0;
+    for (i = 0; i < header_count; i++) {
+        err = dprintf(fd, "%s\r\n", headers[i]);
+        if (err < 0)
+            return -1;
+        sum += err;
+    }
+    return sum;
+}
+
+int write_body(const int fd, const char* body) {
+    int err, sum = 0;
+    err = dprintf(fd, "Content-Length: %ld\r\n\r\n", strlen(body));
+    if (err < 0)
+        return -1;
+    sum += err;
+    err = write(fd, body, strlen(body));
```

Comment:
Do not use `strlen()` here to get rid of possible null characters inside body. 

And please pass the length of `body` from the caller.